--- llvm-15.0.2.src/CMakeLists.txt.orig	2022-10-15 23:14:23.474649716 +0200
+++ llvm-15.0.2.src/CMakeLists.txt	2022-10-15 23:16:11.530931066 +0200
@@ -2,7 +2,7 @@
 
 include(CMakeDependentOption)
 
-set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
+set(LLVM_COMMON_CMAKE_UTILS ${CMAKE_CURRENT_SOURCE_DIR}/../cmake CACHE STRING "Path to CMake modules shared between projects")
 include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
   NO_POLICY_SCOPE)
 
