--- llvm-14.0.4.src/projects/compiler-rt/cmake/base-config-ix.cmake.orig	2022-05-25 00:02:58.000000000 +0200
+++ llvm-14.0.4.src/projects/compiler-rt/cmake/base-config-ix.cmake	2022-05-31 12:22:26.076246743 +0200
@@ -109,13 +109,13 @@
   set(COMPILER_RT_INSTALL_LIBRARY_DIR "${default_install_path}" CACHE PATH
     "Path where built compiler-rt libraries should be installed.")
 endif()
-extend_path(default_install_path "${COMPILER_RT_INSTALL_PATH}" "${CMAKE_INSTALL_BINDIR}")
+extend_path(default_install_path "${COMPILER_RT_INSTALL_PATH}" "bin")
 set(COMPILER_RT_INSTALL_BINARY_DIR "${default_install_path}" CACHE PATH
   "Path where built compiler-rt executables should be installed.")
-extend_path(default_install_path "${COMPILER_RT_INSTALL_PATH}" "${CMAKE_INSTALL_INCLUDEDIR}")
+extend_path(default_install_path "${COMPILER_RT_INSTALL_PATH}" "include")
 set(COMPILER_RT_INSTALL_INCLUDE_DIR "${default_install_path}" CACHE PATH
   "Path where compiler-rt headers should be installed.")
-extend_path(default_install_path "${COMPILER_RT_INSTALL_PATH}" "${CMAKE_INSTALL_DATADIR}")
+extend_path(default_install_path "${COMPILER_RT_INSTALL_PATH}" "share")
 set(COMPILER_RT_INSTALL_DATA_DIR "${default_install_path}" CACHE PATH
   "Path where compiler-rt data files should be installed.")
 
