--- llvm-3.7.0.src/include/llvm/CodeGen/CommandFlags.h~	2015-07-06 19:44:26.000000000 +0200
+++ llvm-3.7.0.src/include/llvm/CodeGen/CommandFlags.h	2015-10-11 11:59:10.130862977 +0200
@@ -21,7 +21,7 @@
 #include "llvm/IR/Intrinsics.h"
 #include "llvm/IR/Module.h"
 #include "llvm/MC/MCTargetOptionsCommandFlags.h"
-#include "llvm//MC/SubtargetFeature.h"
+#include "llvm/MC/SubtargetFeature.h"
 #include "llvm/Support/CodeGen.h"
 #include "llvm/Support/CommandLine.h"
 #include "llvm/Support/Host.h"
