--- llvm-3.9.1.src/tools/lldb/include/lldb/Utility/TaskPool.h~	2015-11-03 23:29:20.000000000 +0100
+++ llvm-3.9.1.src/tools/lldb/include/lldb/Utility/TaskPool.h	2018-02-19 15:48:34.116275467 +0100
@@ -33,6 +33,7 @@
 #include <queue>
 #include <thread>
 #include <vector>
+#include <functional>
 
 // Global TaskPool class for running tasks in parallel on a set of worker thread created the first
 // time the task pool is used. The TaskPool provide no gurantee about the order the task will be run
