--- llvm-3.7.0.src/tools/polly/cmake/polly_macros.cmake~	2015-06-29 21:57:59.000000000 +0200
+++ llvm-3.7.0.src/tools/polly/cmake/polly_macros.cmake	2015-09-09 21:36:43.080128117 +0200
@@ -46,7 +46,7 @@
   endif(MSVC)
   install(TARGETS ${name}
     EXPORT LLVMExports
-    LIBRARY DESTINATION lib
+    LIBRARY DESTINATION lib${LLVM_LIBDIR_SUFFIX}
     ARCHIVE DESTINATION lib${LLVM_LIBDIR_SUFFIX})
   set_property(GLOBAL APPEND PROPERTY LLVM_EXPORTS ${name})
 endmacro(add_polly_library)
