--- llvm-4.0.1.src/tools/clang/lib/Driver/ToolChains.cpp.orig	2017-05-31 12:01:14.000000000 +0200
+++ llvm-4.0.1.src/tools/clang/lib/Driver/ToolChains.cpp	2017-12-10 08:33:21.873800345 +0100
@@ -1549,6 +1549,7 @@
 
   static const char *const X86_64LibDirs[] = {"/lib64", "/lib"};
   static const char *const X86_64Triples[] = {
+      "x86_64-pld-linux", "amd64-pld-linux",
       "x86_64-linux-gnu",       "x86_64-unknown-linux-gnu",
       "x86_64-pc-linux-gnu",    "x86_64-redhat-linux6E",
       "x86_64-redhat-linux",    "x86_64-suse-linux",
@@ -1558,6 +1559,7 @@
   static const char *const X32LibDirs[] = {"/libx32"};
   static const char *const X86LibDirs[] = {"/lib32", "/lib"};
   static const char *const X86Triples[] = {
+      "i686-pld-linux", "i586-pld-linux", "i486-pld-linux", "i386-pld-linux", "athlon-pld-linux",
       "i686-linux-gnu",       "i686-pc-linux-gnu",     "i486-linux-gnu",
       "i386-linux-gnu",       "i386-redhat-linux6E",   "i686-redhat-linux",
       "i586-redhat-linux",    "i386-redhat-linux",     "i586-suse-linux",
@@ -1591,6 +1593,7 @@
 
   static const char *const PPCLibDirs[] = {"/lib32", "/lib"};
   static const char *const PPCTriples[] = {
+      "powerpc-pld-linux",
       "powerpc-linux-gnu", "powerpc-unknown-linux-gnu", "powerpc-linux-gnuspe",
       "powerpc-suse-linux", "powerpc-montavista-linuxspe"};
   static const char *const PPC64LibDirs[] = {"/lib64", "/lib"};
@@ -1603,14 +1606,14 @@
       "powerpc64le-suse-linux", "ppc64le-redhat-linux"};
 
   static const char *const SPARCv8LibDirs[] = {"/lib32", "/lib"};
-  static const char *const SPARCv8Triples[] = {"sparc-linux-gnu",
+  static const char *const SPARCv8Triples[] = {"sparc-pld-linux", "sparc-linux-gnu",
                                                "sparcv8-linux-gnu"};
   static const char *const SPARCv9LibDirs[] = {"/lib64", "/lib"};
-  static const char *const SPARCv9Triples[] = {"sparc64-linux-gnu",
+  static const char *const SPARCv9Triples[] = {"sparc64-pld-linux", "sparc64-linux-gnu",
                                                "sparcv9-linux-gnu"};
 
   static const char *const SystemZLibDirs[] = {"/lib64", "/lib"};
-  static const char *const SystemZTriples[] = {
+  static const char *const SystemZTriples[] = {"s390x-pld-linux",
       "s390x-linux-gnu", "s390x-unknown-linux-gnu", "s390x-ibm-linux-gnu",
       "s390x-suse-linux", "s390x-redhat-linux"};
 
@@ -4112,7 +4115,7 @@
 
   Distro Distro(D.getVFS());
 
-  if (Distro.IsOpenSUSE() || Distro.IsUbuntu()) {
+  if (Distro.IsOpenSUSE() || Distro.IsUbuntu() || Distro.IsPLD()) {
     ExtraOpts.push_back("-z");
     ExtraOpts.push_back("relro");
   }
@@ -4132,7 +4135,7 @@
   // ABI requires a mapping between the GOT and the symbol table.
   // Android loader does not support .gnu.hash.
   if (!IsMips && !IsAndroid) {
-    if (Distro.IsRedhat() || Distro.IsOpenSUSE() ||
+    if (Distro.IsRedhat() || Distro.IsOpenSUSE() || Distro.IsPLD() ||
         (Distro.IsUbuntu() && Distro >= Distro::UbuntuMaverick))
       ExtraOpts.push_back("--hash-style=gnu");
 
@@ -4141,7 +4144,7 @@
       ExtraOpts.push_back("--hash-style=both");
   }
 
-  if (Distro.IsRedhat() && Distro != Distro::RHEL5 && Distro != Distro::RHEL6)
+  if ((Distro.IsRedhat() && Distro != Distro::RHEL5 && Distro != Distro::RHEL6) || Distro.IsPLD())
     ExtraOpts.push_back("--no-add-needed");
 
 #ifdef ENABLE_LINKER_BUILD_ID
--- llvm-4.0.1.src/tools/clang/include/clang/Driver/Distro.h.orig	2016-11-28 22:11:14.000000000 +0100
+++ llvm-4.0.1.src/tools/clang/include/clang/Driver/Distro.h	2017-12-10 08:31:15.017135111 +0100
@@ -27,6 +27,7 @@
     // in this enum, because some tests are done by integer comparison against
     // the first and last known member in the family, e.g. IsRedHat().
     ArchLinux,
+    PLDLinux,
     DebianLenny,
     DebianSqueeze,
     DebianWheezy,
@@ -112,7 +113,11 @@
   bool IsUbuntu() const {
     return DistroVal >= UbuntuHardy && DistroVal <= UbuntuZesty;
   }
- 
+
+  bool IsPLD() const {
+    return DistroVal == PLDLinux;
+  }
+
   /// @}
 };
 
--- llvm-4.0.1.src/tools/clang/lib/Driver/Distro.cpp.orig	2016-11-28 22:11:18.000000000 +0100
+++ llvm-4.0.1.src/tools/clang/lib/Driver/Distro.cpp	2017-12-10 08:32:15.883801082 +0100
@@ -128,6 +128,9 @@
   if (VFS.exists("/etc/arch-release"))
     return Distro::ArchLinux;
 
+  if (VFS.exists("/etc/pld-release"))
+    return Distro::PLDLinux;
+
   return Distro::UnknownDistro;
 }
 
