--- llvm-3.4.1.src/Makefile.rules.orig	2014-05-24 16:58:14.382767108 +0200
+++ llvm-3.4.1.src/Makefile.rules	2014-05-24 20:17:33.645851173 +0200
@@ -1196,7 +1196,7 @@
 	$(Verb) $(INSTALL) $(LibName.SO) $(DestSharedLib)
 ifdef SHARED_ALIAS
 	$(Echo) Creating alias from $(DestSharedLib) to $(DestSharedAlias)
-	$(Verb) $(AliasTool) $(DestSharedLib) $(DestSharedAlias)
+	$(Verb) $(AliasTool) $(BaseLibName.SO) $(DestSharedAlias)
 endif
 
 uninstall-local::
