diff -dur -x '*~' -x '*.orig' -x '*.rej' llvm-3.9.0.src.orig/tools/lld/COFF/CMakeLists.txt llvm-3.9.0.src/tools/lld/COFF/CMakeLists.txt
--- llvm-3.9.0.src.orig/tools/lld/COFF/CMakeLists.txt	2016-07-13 05:19:27.000000000 +0200
+++ llvm-3.9.0.src/tools/lld/COFF/CMakeLists.txt	2016-11-03 12:20:24.000000000 +0100
@@ -18,6 +18,9 @@
   Symbols.cpp
   Writer.cpp
 
+  LINK_LIBS
+    pthread
+
   LINK_COMPONENTS
   ${LLVM_TARGETS_TO_BUILD}
   Core
diff -dur -x '*~' -x '*.orig' -x '*.rej' llvm-3.9.0.src.orig/tools/lld/lib/Core/CMakeLists.txt llvm-3.9.0.src/tools/lld/lib/Core/CMakeLists.txt
--- llvm-3.9.0.src.orig/tools/lld/lib/Core/CMakeLists.txt	2016-01-07 01:14:09.000000000 +0100
+++ llvm-3.9.0.src/tools/lld/lib/Core/CMakeLists.txt	2016-11-03 12:20:10.000000000 +0100
@@ -12,5 +12,5 @@
   ${LLD_INCLUDE_DIR}/lld/Core
 
   LINK_LIBS
-    LLVMSupport
+    LLVMSupport pthread
   )
diff -dur -x '*~' -x '*.orig' -x '*.rej' llvm-3.9.0.src.orig/tools/lld/lib/Driver/CMakeLists.txt llvm-3.9.0.src/tools/lld/lib/Driver/CMakeLists.txt
--- llvm-3.9.0.src.orig/tools/lld/lib/Driver/CMakeLists.txt	2016-03-02 20:08:05.000000000 +0100
+++ llvm-3.9.0.src/tools/lld/lib/Driver/CMakeLists.txt	2016-11-03 12:20:10.000000000 +0100
@@ -17,6 +17,7 @@
     LLVMObject
     LLVMOption
     LLVMSupport
+    pthread
   )
 
 add_dependencies(lldDriver DriverOptionsTableGen)
diff -dur -x '*~' -x '*.orig' -x '*.rej' llvm-3.9.0.src.orig/tools/lld/lib/ReaderWriter/CMakeLists.txt llvm-3.9.0.src/tools/lld/lib/ReaderWriter/CMakeLists.txt
--- llvm-3.9.0.src.orig/tools/lld/lib/ReaderWriter/CMakeLists.txt	2016-03-02 01:37:50.000000000 +0100
+++ llvm-3.9.0.src/tools/lld/lib/ReaderWriter/CMakeLists.txt	2016-11-03 12:20:10.000000000 +0100
@@ -16,4 +16,5 @@
     lldYAML
     LLVMObject
     LLVMSupport
+    pthread
   )
