--- llvm-5.0.0.src.orig/tools/clang/lib/Driver/ToolChains/Gnu.cpp	2016-08-18 19:56:48.000000000 +0200
+++ llvm-5.0.0.src/tools/clang/lib/Driver/ToolChains/Gnu.cpp	2016-11-03 12:18:35.000000000 +0100
@@ -1498,6 +1498,7 @@
 
   static const char *const X86_64LibDirs[] = {"/lib64", "/lib"};
   static const char *const X86_64Triples[] = {
+      "x86_64-pld-linux", "amd64-pld-linux",
       "x86_64-linux-gnu",       "x86_64-unknown-linux-gnu",
       "x86_64-pc-linux-gnu",    "x86_64-redhat-linux6E",
       "x86_64-redhat-linux",    "x86_64-suse-linux",
@@ -1558,6 +1559,7 @@
   static const char *const X32LibDirs[] = {"/libx32"};
   static const char *const X86LibDirs[] = {"/lib32", "/lib"};
   static const char *const X86Triples[] = {
+      "i686-pld-linux", "i586-pld-linux", "i486-pld-linux", "i386-pld-linux", "athlon-pld-linux",
       "i686-linux-gnu",       "i686-pc-linux-gnu",     "i486-linux-gnu",
       "i386-linux-gnu",       "i386-redhat-linux6E",   "i686-redhat-linux",
       "i586-redhat-linux",    "i386-redhat-linux",     "i586-suse-linux",
@@ -1591,6 +1593,7 @@
 
   static const char *const PPCLibDirs[] = {"/lib32", "/lib"};
   static const char *const PPCTriples[] = {
+      "powerpc-pld-linux",
       "powerpc-linux-gnu", "powerpc-unknown-linux-gnu", "powerpc-linux-gnuspe",
       "powerpc-suse-linux", "powerpc-montavista-linuxspe"};
   static const char *const PPC64LibDirs[] = {"/lib64", "/lib"};
@@ -1603,14 +1606,14 @@
       "powerpc64le-suse-linux", "ppc64le-redhat-linux"};
 
   static const char *const SPARCv8LibDirs[] = {"/lib32", "/lib"};
-  static const char *const SPARCv8Triples[] = {"sparc-linux-gnu",
+  static const char *const SPARCv8Triples[] = {"sparc-pld-linux", "sparc-linux-gnu",
                                                "sparcv8-linux-gnu"};
   static const char *const SPARCv9LibDirs[] = {"/lib64", "/lib"};
-  static const char *const SPARCv9Triples[] = {"sparc64-linux-gnu",
+  static const char *const SPARCv9Triples[] = {"sparc64-pld-linux", "sparc64-linux-gnu",
                                                "sparcv9-linux-gnu"};
 
   static const char *const SystemZLibDirs[] = {"/lib64", "/lib"};
-  static const char *const SystemZTriples[] = {
+  static const char *const SystemZTriples[] = {"s390x-pld-linux",
       "s390x-linux-gnu", "s390x-unknown-linux-gnu", "s390x-ibm-linux-gnu",
       "s390x-suse-linux", "s390x-redhat-linux"};
 
--- llvm-5.0.0.src/tools/clang/include/clang/Driver/Distro.h.orig	2017-05-04 14:46:38.000000000 +0200
+++ llvm-5.0.0.src/tools/clang/include/clang/Driver/Distro.h	2017-12-01 22:44:57.785429043 +0100
@@ -27,6 +27,7 @@
     // in this enum, because some tests are done by integer comparison against
     // the first and last known member in the family, e.g. IsRedHat().
     ArchLinux,
+    PLDLinux,
     DebianLenny,
     DebianSqueeze,
     DebianWheezy,
@@ -114,6 +115,10 @@
     return DistroVal >= UbuntuHardy && DistroVal <= UbuntuArtful;
   }
 
+  bool IsPLD() {
+    return DistroVal == PLDLinux;
+  }
+
   /// @}
 };
 
--- llvm-5.0.0.src/tools/clang/lib/Driver/Distro.cpp.orig	2017-05-04 14:46:38.000000000 +0200
+++ llvm-5.0.0.src/tools/clang/lib/Driver/Distro.cpp	2017-12-01 22:47:15.155427475 +0100
@@ -129,6 +129,9 @@
   if (VFS.exists("/etc/arch-release"))
     return Distro::ArchLinux;
 
+  if (VFS.exists("/etc/pld-release"))
+    return Distro::PLDLinux;
+
   return Distro::UnknownDistro;
 }
 
--- llvm-5.0.0.src/tools/clang/lib/Driver/ToolChains/Linux.cpp.orig	2017-12-01 22:14:03.462116884 +0100
+++ llvm-5.0.0.src/tools/clang/lib/Driver/ToolChains/Linux.cpp	2017-12-01 22:51:38.678757801 +0100
@@ -210,7 +210,7 @@
 
   Distro Distro(D.getVFS());
 
-  if (Distro.IsOpenSUSE() || Distro.IsUbuntu()) {
+  if (Distro.IsOpenSUSE() || Distro.IsUbuntu() || Distro.IsPLD()) {
     ExtraOpts.push_back("-z");
     ExtraOpts.push_back("relro");
   }
@@ -232,7 +232,7 @@
   // Android loader does not support .gnu.hash.
   // Hexagon linker/loader does not support .gnu.hash
   if (!IsMips && !IsAndroid && !IsHexagon) {
-    if (Distro.IsRedhat() || Distro.IsOpenSUSE() ||
+    if (Distro.IsRedhat() || Distro.IsOpenSUSE() || Distro.IsPLD() ||
         (Distro.IsUbuntu() && Distro >= Distro::UbuntuMaverick))
       ExtraOpts.push_back("--hash-style=gnu");
 
@@ -241,7 +241,7 @@
       ExtraOpts.push_back("--hash-style=both");
   }
 
-  if (Distro.IsRedhat() && Distro != Distro::RHEL5 && Distro != Distro::RHEL6)
+  if ((Distro.IsRedhat() && Distro != Distro::RHEL5 && Distro != Distro::RHEL6) || Distro.IsPLD())
     ExtraOpts.push_back("--no-add-needed");
 
 #ifdef ENABLE_LINKER_BUILD_ID
