--- llvm-13.0.0.src/tools/flang/CMakeLists.txt.orig	2021-12-19 14:54:33.560125321 +0100
+++ llvm-13.0.0.src/tools/flang/CMakeLists.txt	2021-12-19 14:56:15.709571929 +0100
@@ -196,8 +196,8 @@ else()
   endif()
 
   set(BACKEND_PACKAGE_STRING "${PACKAGE_STRING}")
-  set(MLIR_MAIN_SRC_DIR ${LLVM_MAIN_SRC_DIR}/../mlir/include ) # --src-root
-  set(MLIR_INCLUDE_DIR ${LLVM_MAIN_SRC_DIR}/../mlir/include ) # --includedir
+  set(MLIR_MAIN_SRC_DIR ${LLVM_MAIN_SRC_DIR}/tools/mlir/include ) # --src-root
+  set(MLIR_INCLUDE_DIR ${LLVM_MAIN_SRC_DIR}/tools/mlir/include ) # --includedir
   set(MLIR_TABLEGEN_OUTPUT_DIR ${CMAKE_BINARY_DIR}/tools/mlir/include)
   set(MLIR_TABLEGEN_EXE $<TARGET_FILE:mlir-tblgen>)
   include_directories(SYSTEM ${MLIR_INCLUDE_DIR})
