--- llvm-13.0.0.src/cmake/config-ix.cmake.orig	2021-09-24 18:18:10.000000000 +0200
+++ llvm-13.0.0.src/cmake/config-ix.cmake	2021-11-08 11:27:55.475562698 +0100
@@ -625,7 +625,7 @@
 set(LLVM_BINDINGS "")
 include(FindOCaml)
 include(AddOCaml)
-if(WIN32 OR NOT LLVM_ENABLE_BINDINGS)
+if(WIN32 OR NOT LLVM_ENABLE_BINDINGS OR (DEFINED LLVM_BINDINGS_LIST AND NOT "ocaml" IN_LIST LLVM_BINDINGS_LIST))
   message(STATUS "OCaml bindings disabled.")
 else()
   find_package(OCaml)
