--- llvm-13.0.0.src/tools/clang/lib/Driver/ToolChains/Gnu.cpp.orig	2021-10-25 20:48:42.045696956 +0200
+++ llvm-13.0.0.src/tools/clang/lib/Driver/ToolChains/Gnu.cpp	2021-10-25 20:56:43.802012798 +0200
@@ -2113,8 +2113,9 @@
       "x86_64-manbo-linux-gnu", "x86_64-linux-gnu",
       "x86_64-slackware-linux", "x86_64-unknown-linux",
       "x86_64-amazon-linux"};
-  static const char *const X32Triples[] = {"x86_64-linux-gnux32",
-                                           "x86_64-pc-linux-gnux32"};
+  static const char *const X32Triples[] = {
+      "x86_64-pld-linux-gnux32", "amd64-pld-linux-gnux32",
+      "x86_64-linux-gnux32", "x86_64-unknown-linux-gnux32", "x86_64-pc-linux-gnux32"};
   static const char *const X32LibDirs[] = {"/libx32", "/lib"};
   static const char *const X86LibDirs[] = {"/lib32", "/lib"};
   static const char *const X86Triples[] = {
@@ -2352,12 +2353,16 @@
       LibDirs.append(begin(X32LibDirs), end(X32LibDirs));
       TripleAliases.append(begin(X32Triples), end(X32Triples));
       BiarchLibDirs.append(begin(X86_64LibDirs), end(X86_64LibDirs));
+      BiarchLibDirs.append(begin(X86LibDirs), end(X86LibDirs));
       BiarchTripleAliases.append(begin(X86_64Triples), end(X86_64Triples));
+      BiarchTripleAliases.append(begin(X86Triples), end(X86Triples));
     } else {
       LibDirs.append(begin(X86_64LibDirs), end(X86_64LibDirs));
       TripleAliases.append(begin(X86_64Triples), end(X86_64Triples));
       BiarchLibDirs.append(begin(X32LibDirs), end(X32LibDirs));
+      BiarchLibDirs.append(begin(X86LibDirs), end(X86LibDirs));
       BiarchTripleAliases.append(begin(X32Triples), end(X32Triples));
+      BiarchTripleAliases.append(begin(X86Triples), end(X86Triples));
     }
     BiarchLibDirs.append(begin(X86LibDirs), end(X86LibDirs));
     BiarchTripleAliases.append(begin(X86Triples), end(X86Triples));
--- llvm-15.0.2.src/projects/compiler-rt/cmake/Modules/CompilerRTUtils.cmake.orig	2022-10-04 11:29:17.000000000 +0200
+++ llvm-15.0.2.src/projects/compiler-rt/cmake/Modules/CompilerRTUtils.cmake	2022-10-16 22:33:08.014906010 +0200
@@ -478,7 +478,8 @@
 
 function(get_compiler_rt_target arch variable)
   string(FIND ${COMPILER_RT_DEFAULT_TARGET_TRIPLE} "-" dash_index)
-  string(SUBSTRING ${COMPILER_RT_DEFAULT_TARGET_TRIPLE} ${dash_index} -1 triple_suffix)
+  string(SUBSTRING ${COMPILER_RT_DEFAULT_TARGET_TRIPLE} ${dash_index} -1 triple_suffix_with_abi)
+  string(REPLACE "-gnux32" "" triple_suffix ${triple_suffix_with_abi})
   if(COMPILER_RT_DEFAULT_TARGET_ONLY)
     # Use exact spelling when building only for the target specified to CMake.
     set(target "${COMPILER_RT_DEFAULT_TARGET_TRIPLE}")
